src/ragas/executor.py

```diff
@@ -114,29 +48,53 @@ def submit(
         self, callable: t.Callable, *args, name: t.Optional[str] = None, **kwargs
     ):
         callable_with_index = self.wrap_callable_with_index(callable, len(self.jobs))
-        self.jobs.append((callable_with_index(*args, **kwargs), name))
+        self.jobs.append((callable_with_index, args, kwargs, name))
 
     def results(self) -> t.List[t.Any]:
-        executor_job = Runner(
-            jobs=self.jobs,
-            desc=self.desc,
-            keep_progress_bar=self.keep_progress_bar,
-            raise_exceptions=self.raise_exceptions,
-            run_config=self.run_config,
+        loop = asyncio.get_event_loop()
```

Comment:
[Use current loop if available](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop) so we're not creating unnecessary new loops that seems to cause tasks to be on different loops problem

Reply:
![image](https://github.com/explodinggradients/ragas/assets/5261489/9f3ba4de-cbd3-4a31-901d-26af59eec05e)
this is why I planned not to use `get_event_loop()` - would have been much easier if we could :(

what do you think?

Reply:
Oh I didn't know that.
With my shuffling of job creation vs. new loop order, I think theoretically `new_event_loop()` should also work.
Will test and report.

Reply:
`new_event_loop()` works too and basically creates a new Thread I believe. running a new Thread/event loop is the cleanest implementation because if we reuse the same event loop in Jupiter notebook - it causes a lot of issues.

alternatively, what was the main problem with the current setup? if it's traces we can improve that part actually (traces and harder to debug were the points against this)

Reply:
Since I was getting #681 I thought the issue was that in the current setup futures were created before the the new loop was created, and later the semaphore refers to the new loop but futures (rather, ‘jobs’) were already attached to a old, different loop.

Furthermore, ‘Runner’ itself is a new thread, so doesn’t it mean the new loop exists on a new thread whereas the ‘jobs’ were created for on a loop in the main thread?

Please correct me if I’m wrong. The fact that sometimes the code works without this patch is a mystery to me.